src/lib/utils.ts

```diff
@@ -153,7 +153,12 @@ export function toSync<F extends AsyncFn<F>>(
   ) => void | PromiseLike<void | null | undefined> | null | undefined
 ): (...args: Parameters<F>) => void {
   return (...args: Parameters<F>) => {
-    fn(...args).catch(onReject)
+    try {
+      void fn(...args)
+    } catch(e) {
+      console.trace()
+      void onReject(e)
+    }
```

Comment:
`fn` is required to be async by the type. By using `void` when calling it, as in `void fn(...args)`, you're firing it off and not awaiting it. So this undoes the showing of errors in `fn`.

Reply:
I think this is what you mean to do.

```suggestion
    void fn(...args).catch((err) => {
      console.trace()
      void onReject(err)
    })
```

Reply:
No I've tried this before, and it clobbers the trace.

In this suggestion you're still not awaiting the promise.

In the original case, we were not awaiting the promise: notice there is no return. We were simply catching. `try{}catch{}` here has the exact same behavior.

Reply:
I stand corrected: while the original change did to something in order to show stack traces, it was made clear to me I still need to `await` in order for `try{}catch{}` to properly work here.

After discussion a bit further, I remembered I could do `new Error` to capture a stack and display this in the `catch` :) Thanks Jon!